orca-core/src/main/java/com/netflix/spinnaker/orca/notifications/scheduling/OldPipelineCleanupPollingNotificationAgent.java

```diff
@@ -130,21 +137,43 @@ private void startPolling() {
   }
 
   private void tick() {
-    ScanParams sp = new ScanParams().match("pipeline:app:*").count(2000);
     String cursor = "0";
     try {
       List<String> applications = new ArrayList<>();
+
+      CursorBasedResult<String> dynoResult = null;
       while (true) {
         String tmpCursor = cursor;
-        ScanResult<String> result = withJedis(jedis -> jedis.scan(tmpCursor, sp));
-        cursor = result.getStringCursor();
 
-        // pipeline:app:foo -> foo
-        applications.addAll(result.getResult().stream().map(k -> k.split(":")[2]).collect(Collectors.toList()));
-
-        if ("0".equals(cursor)) {
-          break;
+        if (redisClientDelegate.supportsTransactions()) {
+          ScanParams sp = new ScanParams().match(scanPattern()).count(2000);
+          ScanResult<String> result = redisClientDelegate.withMultiClient(client -> {
+            return client.scan(tmpCursor, sp);
+          });
+          cursor = result.getStringCursor();
+
+          // pipeline:app:foo -> foo
+          applications.addAll(result.getResult().stream().map(k -> k.split(":")[2]).collect(Collectors.toList()));
+          if ("0".equals(cursor)) {
+            break;
+          }
+        } else {
+          // Dynomite
+          final CursorBasedResult<String> lastResult = objectMapper.readValue(
+            objectMapper.writeValueAsBytes(dynoResult), new TypeReference<CursorBasedResult<String>>() {
```

Comment:
There has to be a better way to do this without running afoul of the rule against non-final local variables in lambdas.